store/sqlstore/user_store.go

```diff
@@ -1361,3 +1361,466 @@ func (us SqlUserStore) InferSystemInstallDate() store.StoreChannel {
+				        AND TargetInteraction.UserId = :uid
+				        AND Parent_Post_Author_Id = :uid
+				        AND ChannelMembersUsers.Id != :uid)
+				)
+			) AS UserInteractions
+
+			WHERE Username IN (SELECT Username -- all interactions must take place with users on your team
+			                          FROM Users
+
+			                            JOIN TeamMembers
+			                            ON TeamMembers.TeamID = :teamId
+			                            AND TeamMembers.UserId = Users.Id
+			                        )
+			`
+			var userInteractions []*model.UserInteraction
+
+			if _, err :=  us.GetReplica().Select(&userInteractions, query, map[string]interface{}{"uid": uid, "teamId": teamId}); err != nil {
+				result.Err = model.NewAppError("SqlUserStore.GetUserInteractions", "store.sql_post.ge_user_interactions.app_error", nil, err.Error(), http.StatusInternalServerError)
+			}
+
+			result.Data = userInteractions
+	})
+}
+
+func (us *SqlUserStore) GetUserLearningGroups(uid string, teamId string, learningGroupPrefixes []*model.LearningGroupPrefix) store.StoreChannel {
+	// For every learning group prefix for this team, use this case statement
+	// to record a name for this channel(team), minus the prefix (ex. plg-30 => 30)
+	prefixReplaceQuery := ""
+
+	for _, prefix := range learningGroupPrefixes {
+		prefixReplaceQuery += `
+		WHEN LEFT(Channels.Name,` + strconv.Itoa(len(prefix.Prefix) + 1) + `) = '` + prefix.Prefix + `-'
+		THEN REPLACE(Channels.Name,'` + prefix.Prefix + `-','')
+		`
+	}
+
+	// If there are no learning group prefixes, then '' as team name,
+	// since this should be a 1 to 1 relationship
+	if len(prefixReplaceQuery) == 0 {
+		prefixReplaceQuery = " WHEN 1 = 1 THEN '' "
+	}
+
+	// For every learning group prefix for this team, use this case statement
+	// to get a team type for this channel(team) (ex. plg-30 => plg)
+	prefixQuery := ""
+
+	for _, prefix := range learningGroupPrefixes {
+		prefixQuery += `
+		WHEN LEFT(Channels.Name,` + strconv.Itoa(len(prefix.Prefix) + 1) + `) = '` + prefix.Prefix + `-'
+		THEN '` + prefix.Prefix + `'
+		`
+	}
+
+	// If there are no learning group prefixes, then '' as team type,
+	// since this should be a 1 to 1 relationship
+	if len(prefixQuery) == 0 {
+		prefixQuery = " WHEN 1 = 1 THEN '' "
+	}
+
+
+	// For every learning group prefix for this team, use this case statement
+	// to find all channels with a name that starts with the prefix
+	prefixFilterQuery := ""
+
+	for _, prefix := range learningGroupPrefixes {
+		if len(prefixFilterQuery) > 0 {
+			prefixFilterQuery += " OR "
+		}
+		prefixFilterQuery += `
+		LEFT(Channels.Name,` + strconv.Itoa(len(prefix.Prefix) + 1) + `) = '` + prefix.Prefix + `-'
+		`
+	}
+
+	// If there are no learning group prefixes, then use 1 != 1 to return no results,
+	// since no prefixes means no learning group channels
+	if len(prefixFilterQuery) == 0 {
+		prefixFilterQuery = " 1 != 1 "
+	}
+
+	return store.Do(func(result *store.StoreResult) {
```

Comment:
And here, I'm again convinced that we want a Query builder, or at least a lot of documentation.